cmd/search/views/views.go

```diff
@@ -0,0 +1,122 @@
+package views
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os"
+)
+
+// These constants represent the different layouts in use.
+const (
+	RESULTS = "Results"
+	SEARCH  = "Search"
+	LAYOUT  = "Layout"
+)
+
+// Options ...
+type Options struct {
+	Term  string
+	CNN   bool
+	NYT   bool
+	BBC   bool
+	First bool
+}
+
+// Result ...
+type Result struct {
+	Engine  string
+	Title   string
+	Link    string
+	Content string
+}
+
+// TitleHTML fixes encoding issues. The templates expect this
+// method for rendering.
+func (r *Result) TitleHTML() template.HTML {
+	return template.HTML(r.Title)
+}
+
+// ContentHTML fixes encoding issues. The templates expect this
+// method for rendering.
+func (r *Result) ContentHTML() template.HTML {
+	return template.HTML(r.Content)
+}
+
+// views contains a map of static templates for rendering views.
+var views = make(map[string]*template.Template)
+
+// Render generates the HTML response for this route.
+func Render(fv map[string]interface{}, results interface{}) ([]byte, error) {
+	var markup bytes.Buffer
+
+	// Generate the markup for the results template.
+	if results != nil {
+		vars := map[string]interface{}{"Items": results}
+		if err := views[RESULTS].Execute(&markup, vars); err != nil {
+			return nil, fmt.Errorf("error processing results template : %v", err)
+		}
+		fv[RESULTS] = template.HTML(markup.String())
+	}
+
+	// Generate the markup for the search template.
+	markup.Reset()
+	if err := views[SEARCH].Execute(&markup, fv); err != nil {
+		return nil, fmt.Errorf("error processing search template : %v", err)
+	}
+
+	// Generate the final markup with the layout template.
+	vars := map[string]interface{}{"LayoutContent": template.HTML(markup.String())}
+	markup.Reset()
+	if err := views[LAYOUT].Execute(&markup, vars); err != nil {
```

Comment:
`html/template` provides support for nesting blocks of templates together so we only need one call to `*template.Execute`. I will look into this but it doesn't need to hold up the work on this PR.

Reply:
Please do, that would be amazing.